pkg/iter/iter.go

```diff
@@ -0,0 +1,44 @@
+// Copyright (c) 2024 The Jaeger Authors.
+// SPDX-License-Identifier: Apache-2.0
+
+// Package iter is a backport of Go 1.23 official "iter" package, until we upgrade.
+package iter
+
+type (
+	Seq[V any]     func(yield func(V) bool)
+	Seq2[K, V any] func(yield func(K, V) bool)
+)
+
+func CollectWithErrors[V any](seq Seq2[V, error]) ([]V, error) {
```

Comment:
Why are we calling this `CollectWithErrors` instead of just `Collect`?

Reply:
Because it's not a plain collect, which would be defined over Seq not Seq2